src/chain.h

```diff
@@ -378,6 +378,7 @@ class CChain {
 
     /** Efficiently check whether a block is present in this chain. */
     bool Contains(const CBlockIndex *pindex) const {
+        if (!pindex) return false;  // null pointer isn't in this chain
```

Comment:
dgenr8 or be refactored to be references as well. I always try to use references until it's truly an optional value